### DISCUSSION TITLE
Bindless Rendering

### DISCUSSION THREAD

**Discussion by JonahGoldsmith** [KhronosGroup/MoltenVK#2135] [Q&A] [answered]
Does MoltenVK support unbounded arrays in the shader code along side using the bindless model?

**Comment by billhollings** [comment, marked as answer]
> Does MoltenVK support unbounded arrays in the shader code along side using the bindless model?

Unbounded arrays of primitives or structs in a buffer will definitely work.

Unbounded arrays of resources (textures, samplers ,buffers) in the shader will work as long as the pipeline descriptor layout defines the array size.

**Reply by JonahGoldsmith** [reply]
Thanks!
